### DISCUSSION TITLE
How to add pagination to category index page?

### DISCUSSION THREAD

**Discussion by limakzi** [squidfunk/mkdocs-material#5249] [Q&A] [answered]
Recently, I have been very happy to see a [blog plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#setting-up-a-blog) released. Blog module has a [`categories` feature](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#adding-categories).  Is it possible to add [pagination](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#pagination) to category index page as it is done in blog posts index page?

**Comment by squidfunk** [comment]
It's currently not supported, because the initial assumption was that categories themselves will probably not contain that many entries as the general blog index page. However, we could add pagination to category and archive index pages if that's necessary. You can create a change request if you like, so we can investigate adding this in the next iteration of the blog plugin.

**Reply by strausmann** [reply]
@squidfunk 
we have a problem with the pagination feature. For example, we have a category "2023" and a tag "Heads Up".
When a category or tag gets into the situation of having so many articles that a pagination is required, one is forced to create in the blog folder the folder structure "blog/2023/page/2.md" and "blog/heads-up/page/2.md".

I'm not quite sure now if it's due to both (category/tag) since we use 2023 on as the tag.

We obviously only come across this problem when we run a build with the --strict parameter.

**Reply by strausmann** [reply]
It occurs to me that we have also had this problem when the number of blog posts also leads to a 2nd page so a pagination. Here we had to create under "blog/page/2.md".

We have renamed blog to circulars. This is the error message we get. This is probably also due to the use of "git-authors-plugin".

![image](https://github.com/squidfunk/mkdocs-material/assets/4519228/66bf81da-0dc9-471d-9fed-4ce29fe69910)

**Reply by squidfunk** [reply]
Thanks for reporting. Could you please create a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/), so we can look into it?

**Reply by strausmann** [reply]
I have minimally recreated the project here. To trigger the error, I use this command for a build.

mkdocs build --strict --verbose --site-dir test

https://github.com/strausmann/mkdocs-material-bugreport

**Reply by strausmann** [reply]
@squidfunk here the warnings

![image](https://github.com/squidfunk/mkdocs-material/assets/4519228/7802a855-c0b7-41c5-a874-8c57f36574c6)

**Reply by squidfunk** [reply]
Okay, so if you're purely referring to the warning by the git-authors plugin, please upstream this bug report to the maintainers of the plugin, because the plugin needs to exclude generated pages. All pages that are generated are marked by the plugin:

https://github.com/squidfunk/mkdocs-material/blob/a0206fd34eafa5c986b5d9d27b5173aa1436b3c6/src/plugins/blog/plugin.py#L862-L867

There's also an open PR over at MkDocs to upstream this field into core, so that plugins can better discern generated files from existing files, but I don't expect this to be merged soon, given how controversial this change is regarded by the maintainers.

- https://github.com/mkdocs/mkdocs/pull/3344

If you're referring to another error that I'm not seeing, please create an issue and attach the minimal reproduction by following our regular bug reporting process – we'll look into it then.

**Reply by squidfunk** [reply]
@strausmann btw, please don't hijack discussions that are not related to your error. It makes discovery for other users harder if they run into the same problem. We also mentioned this in [our guidelines](https://squidfunk.github.io/mkdocs-material/contributing/#before-asking-a-question):

> ### Before asking a question
>
> - Is the topic a question for our [discussion board](https://github.com/squidfunk/mkdocs-material/discussions), or is it a bug report or change request that should better be raised on our [issue tracker](https://github.com/squidfunk/mkdocs-material/issues)?
>
> - Is there an open discussion on the topic of your request? If the answer is yes, does your question match the direction of the discussion, or should you open a new discussion?
> 
> - Did your provide our community with all the necessary information to understand your question and help you quickly, or can you make it easier to help you?

**Comment by squidfunk** [comment, marked as answer]
As of [9.2.0](https://squidfunk.github.io/mkdocs-material/changelog/#9.2.0), categories and archive pages are now paginated exactly as the blog index. No configuration necessary ☺️

**Reply by strausmann** [reply]
Unfortunately, this is not true. We are forced to create a 2...n.md. A 1.md, on the other hand, is not required. But because a 2...n.md is required, this always disturbs the automatic build process after 10 new posts, because the colleagues do not think of that. And since a 1.md is not required, it looks to me like a bug.

**Reply by squidfunk** [reply]
As said in https://github.com/squidfunk/mkdocs-material/discussions/5249#discussioncomment-7081254, this is likely a problem with the git-authors plugin.
